### PR TITLE
Use gopher group in CODEOWNERS instead of enumerating its members

### PR DESCRIPTION
**Pull request by wozniakjan**
https://github.com/orgs/kyma-project/teams/gopher/members

similar to https://github.com/kyma-project/control-plane/pull/2492

### CODE REVIEW

**Review by wozniakjan — COMMENTED** [review]
/hold

**Review by PK85 — APPROVED** [review]

File: CODEOWNERS [hunk 1]
```diff
@@ -44,10 +42,10 @@
 /prow/jobs/kyma/components/istio-installer @tomasz-smelcerz-sap @werdes72 @pk85 @dariusztutaj @strekm @kyma-project/prow
 /prow/jobs/kyma/components/kyma-operator @tomasz-smelcerz-sap @werdes72 @pk85 @dariusztutaj @strekm @kyma-project/prow
 /prow/jobs/kyma/components/permission-controller @pk85 @dariusztutaj @strekm @tomasz-smelcerz-sap @werdes72 @kyma-project/prow
-/prow/jobs/kyma/components/telemetry-operator @pk85 @szwedm @tillknuesting @a-thaler @ralikio @wozniakjan @piotrmiskiewicz @kyma-project/prow
-/prow/jobs/kyma/components/directory-size-exporter @pk85 @szwedm @tillknuesting @a-thaler @ralikio @wozniakjan @piotrmiskiewicz @kyma-project/prow
+/prow/jobs/kyma/components/telemetry-operator @a-thaler @kyma-project/prow
```
Comment (halamix2) [inline CODEOWNERS:45]: Should kyma-team/gopher be removed here?

Comment (wozniakjan) [reply CODEOWNERS:45]: afaict yes, asked the whole @kyma-project/gopher team and no one knows what it is and looks like nobody from the gopher team ever contributed there
```
$ git log --pretty=format:%an prow/jobs/kyma/components/telemetry-operator/ | sort | uniq
Christoph Kleineweber
Dennis Geiselmann
Hisar Balik
Kamil Kasperski
Kyma Bot
Piotr Halama
Przemek Pokrywka
```

Comment (Sawthis) [reply CODEOWNERS:45]: @a-thaler it looks like you are the last owner from development teams for these 2 jobs. Are these jobs still needed?

**Review by Sawthis — APPROVED** [review]

**Pull request merged**